### DISCUSSION TITLE
Parallel Fork implementation

### DISCUSSION THREAD

**Discussion by brendt** [spatie/async#143] [Ideas]
I was playing around with a Laravel project when I needed to run some code in parallel. I didn't want to use `spatie/async` because you're required to manually bootstrap your child processes. That made me think about a potential new feature, one that wraps around `pcntl_fork`, which don't require manual bootstraps since these child processes are forks of the parent. 

Forks are slower, but if all you need is to run some code in parallel, they might be a good alternative. Here's an example of my test case, the syntax would be very minimal, with the only goal to run a few callables in parallel: 

```php
$bus = app(CommandBus::class);

$result = Fork::new()
    ->before(fn () => DB::connection('mysql')->reconnect()) // Needed for Laravel to properly communicate with the DB in forks
    ->run(
        fn () => $bus->dispatch(new AddItem('cart-uuid', 'item-1')),
        fn () => $bus->dispatch(new AddItem('cart-uuid', 'item-1')),
    );
```

**Comment by skrtdev** [comment]
Creators of spatie/async say that they don’t like forking. I created a library that uses forking cause I needed that for my telegram bot library. Check it out at https://github.com/skrtdev/php-async. One of the cons of forking is that you can’t have the result of the process

**Reply by brendt** [reply]
I am the creator of spatie/async 😅

For the record: spatie/async doesn't use process forks because using separate processes offer more flexibility. But on the other hand, sometimes you just need a small function to run stuff in parallel, forks can be useful there.

FYI we're going to create a new package that wraps around `pcntl_fork` for easier use, and leave spatie/async for the more complex stuff.

**Reply by skrtdev** [reply]
Oh ok. However there isn't much to wrap, as you can see in my library
